codegen/winterfell/src/air/graph.rs

```diff
@@ -30,15 +32,15 @@ impl Codegen for IntegrityConstraintDegree {
 }
 
 impl Codegen for NodeIndex {
-    fn to_string(&self, ir: &AirIR, elem_type: ElemType) -> String {
+    fn to_string(&self, ir: &AirIR, elem_type: ElemType, trace_segment: TraceSegment) -> String {
         let op = ir.constraint_graph().node(self).op();
-        op.to_string(ir, elem_type)
+        op.to_string(ir, elem_type, trace_segment)
     }
 }
 
 impl Codegen for Operation {
     // TODO: Only add parentheses in Add and Mul if the expression is an arithmetic operation.
-    fn to_string(&self, ir: &AirIR, elem_type: ElemType) -> String {
+    fn to_string(&self, ir: &AirIR, elem_type: ElemType, trace_segment: TraceSegment) -> String {
```

Comment:
Why do we need to pass this as a parameter here? It seems like it is only used in `Operation::TraceElement(trace_access) => {` branch and there `trace_access` already contains `trace_segment` info. Or am I missing something?

Reply:
This is the trace segment of the overall constraint instead of just that particular `trace_access` which would be different for aux transition constraints which access main trace columns. I think we can remove this if we are ok with changing the name of the main evaluation frame in "evaluate_transition" function from "frame" to "main_frame" and handle it the same as "evaluate_aux_transition" (i.e. `main_frame.current()[]`) but I didn't know if we wanted to change the name.

Reply:
I think changing the name is better than passing an extra variable around.

Eventually, we should probably declare local variables like `main_current`, `main_next`, `aux_current`, `aux_next` - but this is probably for another PR.

Reply:
That would be great! Would it be possible to output `aux_current` and `aux_next` only if we have auxiliary constraints? (i.e., when the number of segments is greater than 1)

Reply:
Done.